htlcswitch/iterator.go

```diff
@@ -175,15 +176,9 @@ func (r *sphinxHopIterator) ForwardingInstructions() (ForwardingInfo, error) {
 		var cid uint64
 
 		tlvStream, err := tlv.NewStream(
-			tlv.MakeDynamicRecord(
-				tlv.AmtOnionType, &amt, nil,
-				tlv.ETUint64, tlv.DTUint64,
-			),
-			tlv.MakeDynamicRecord(
-				tlv.LockTimeOnionType, &cltv, nil,
-				tlv.ETUint32, tlv.DTUint32,
-			),
-			tlv.MakePrimitiveRecord(tlv.NextHopOnionType, &cid),
+			record.NewAmtToFwdRecord(&amt),
+			record.NewLockTimeRecord(&cltv),
+			record.NewNextHopIDRecord(&cid),
```

Comment:
Minor comment: as a reviewer, I'd prefer this and the previous commit to be squashed so that it makes a logical unit of work and I can compare that it is indeed just a code move. Now looking at the first commit raises the question what is going to happen next and I see myself going back and forth between these commits.

But it isn't important and subjected to personal preference too.